language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/contexts/BallerinaCompletionContextImpl.java

```diff
@@ -82,4 +87,32 @@ public void addResolver(Node node) {
     public List<Node> getResolverChain() {
         return this.resolverChain;
     }
+
+    @Override
+    public Optional<TypeSymbol> getContextType() {
+
+        if (this.isContextTypeFound) {
+            return Optional.ofNullable(this.contextType);
+        }
+
+        this.isContextTypeFound = true;
+        if (this.getNodeAtCursor() == null) {
+            this.contextType = null;
+            return Optional.ofNullable(contextType);
+        }
+
+        NonTerminalNode node = getNodeAtCursor();
+        do {
+            ContextTypeResolver contextTypeResolver = new ContextTypeResolver(this);
+            Optional<TypeSymbol> typeSymbol = node.apply(contextTypeResolver);
+            if (typeSymbol == null || typeSymbol.isEmpty()) {
+                this.contextType = null;
+            } else {
+                this.contextType = typeSymbol.get();
+            }
+            node = node.parent();
+        } while (this.contextType == null && node.parent() != null);
```

Comment:
Shall we add a null check for the node as well?

Reply:
done, Ideally, the null check should be for the node not the  node.parent()